text/0000-standard-dependency-metadata-format.md

```diff
@@ -0,0 +1,67 @@
+# Standardized Paketo Dependency Metadata Format
+
+## Summary
+
+This proposal suggests that the Paketo project should converge on a single
+dependency metadata format.
+
+## Motivation
+
+There are several projects, such as the introduction of ARM or the removal of
+stacks, that are being discussed that would force Paketo to update the metadata
+for dependencies. It seems prudent that if we are going to have to do a large
+update to dependency metadata that it might also be a good time for us to
+converge as a project on the dependency metadata format that we should be using
+going forward.
+
+This could also be a good first step in converging on shared tooling. By having
+a shared dependency metadata format we would have a good common convergence
+point to begin building universal tooling for the Paketo project.
+
+## Implementation
+
+The following is the proposed metadata format:
+
+```toml
+[[metadata.dependencies]]
+checksum = "<dependency algo:checksum>"
+id = "<dependency ID>"
+uri = "<dependency URI>"
+version = "dependency version"
+
+arch = "<dependency compatible architecture>" #optional
+cpes = [ "<dependency cpe>" ] #optional
+eol-date = "<dependency eol>" #optional
+name = "<dependency name>" #optional
+os = "<dependency compatible OS>" #optional
+purl = "<dependency purl>" #optional
```

Comment:
Why do we support multiple `cpes`, but only a single `purl`?

Reply:
Makes sense to align this with `cpes` even if only one entry is added for most dependencies.

Reply:
Pardon this question, but does anyone actually use PURLs?  I know scanners and users do look at CPEs. I've never heard anyone mention PURL besides asking what it is. Before we roll out the red carpet for PURL, I just want to get a feel for how much people use it.

It's not a huge pain to support, but when you talk about changing things with it, that's going to require work. Makes me want to understand the usage first before committing to changing things.

Reply:
For example https://osv.dev uses purls. More broadly speaking, we should adapt to whatever e.g. security researchers use for reporting vulnerabilities. For a starter, I think that means we should allow a package to have multiple names. I don't see any further package naming scheme coming besides CPE and pURL - but if a relevant new scheme comes up, we should adopt that as well.

Granted, it also depends on using the "correct" names, which might turn out to be difficult as well. Most of the current pURLs used for example are unlikely to match vulnerabilities. I think I brought that up in the past and @ForestEckhardt  (and @sophiewigmore?) wanted to maybe spend some thought and look into that topic.

Reply:
Current pURL implementation context: When added pURLs we were trying to cover all of our bases while we were still trying to figure out emerging trends in how people track packages/CVEs. Our best guess at that point was to leverage https://github.com/package-url/packageurl-go while getting dependency metadata together.

Fast forward to now - I have heard 0 feedback on if they're useful fields or if any useful metadata can be gleaned from the pURLs we provide now. We haven't invested any time as far as I know into getting an answers in this area. I think its likely still a useful field to provide for the same reasons Jan mentioned.

Reply:
Unless someone is going to commit to going through the metadata and cleaning up the PURLs, I feel like we should drop that metadata. 

My understanding of why we adopted them in the first place matches what @sophiewigmore mentioned. We were basically throwing things out there to see what would stick. PURLs don't seem to have stuck, so we should drop the PURL field rather than continue publishing something misleading and likely incorrect.

That is unless someone feels strongly enough about keeping it to review and update all the PURLs that we publish.

Reply:
I agree that the worst would be to stick to the existing `generic` purls that very likely don't help in finding vulnerabilities.

There seem to be [some adopters](https://github.com/package-url/purl-spec#users-adopters-and-links) though, amongst them Google and Github, I wonder if we could also opt to keep purls (plural!) in as optional metadata (and making CPEs optional as well maybe?) - but remove all the `generic` purls we have currently assigned.

One example would be the Apache Tomcat buildpack. While the distribution that gets installed is different from the Tomcat Jar that is available as Maven GAV - there can be vulnerabilities for it found in osv.dev based on the `maven` purl and it might make sense for this buildpack to include the purl as well for this specific reason.

Reply:
I have made the field `purls` and made it so that multiple PURLS can be provided.